scripts/create_static_narrative.py

```diff
@@ -0,0 +1,203 @@
+        results = ws_client.list_objects(
+            {ws_args: [ws_ref], "type": "KBaseNarrative.Narrative"}
+        )
+        if results[0] and results[0][0]:
+            return f"{results[0][6]}/{results[0][0]}/{results[0][4]}"
+
+        # no narrative object
+        msg = f"Workspace {ws_ref} did not contain a KBaseNarrative.Narrative object."
+        raise ValueError(msg)
+
+    def create_static_narrative(
+        self: "StaticNarrativeCmdLine",
+        ws_ref: str | int,
+        user_id: str,
+        token: str,
+        skip_permissions_checks: str,
+    ) -> None:
+        """Create a static narrative from a Workspace reference.
+
+        :param ws_ref: workspace reference; may be the narrative UPA or a workspace ID
+        :type ws_ref: str
+        :param user_id: valid KBase user ID
+        :type user_id: str
+        :param token: KBase token, valid for whichever server the workspace is on
+        :type token: str
+        :param skip_permissions_checks: whether the permission checks should be run
+        :type skip_permissions_checks: str
+        """
+        # this is a workspace reference
+        if ws_ref.count("/") == 0:
+            narrative_ref = self.get_narrative_id_from_workspace(ws_ref, token)
+        else:
+            narrative_ref = ws_ref
+
+        ref = NarrativeRef.parse(narrative_ref)
+        snc = StaticNarrativeCreator(self.config)
+
+        log_msg = f"Creating Static Narrative {ref}"
+        self.logger.info(log_msg)
+
+        if not skip_permissions_checks:
+            snc.check_permissions(ref, user_id=user_id, token=token)
+
+        output_path = snc.export_narrative(ref, user_id, token)
+        log_msg = f"Static Narrative for {ref} created at {output_path}"
+        self.logger.info(log_msg)
+
+
+def parse_args(args: list[str]) -> dict[str, Any]:
+    """Parse input arguments.
+
+    :param args: input argument list
+    :type args: list[str]
+    :raises ValueError: if one or more of the parameters are missing
+    :return: parsed arguments
+    :rtype: dict[str, str]
+    """
+    p = argparse.ArgumentParser()
+    p.add_argument("-u", "--user", dest="user_id", default=None, help="User ID")
+    p.add_argument("-t", "--token", dest="token", default=None, help="User auth token")
+    p.add_argument("-e", "--env", dest="env", default=None, help="KBase environment")
+    p.add_argument(
+        "-w", "--ws", dest="ws_id", default=None, help="Workspace ID with Narrative"
+    )
+    p.add_argument(
+        "-o", "--outdir", dest="outdir", default=".", help="File output directory"
+    )
+    p.add_argument(
+        "-x",
+        "--skip-permissions-checks",
+        dest="skip_permissions_checks",
+        default=None,
+        help="Skip the workspace permissions checks; omit the argument to ensure that permission checks are run.",
+    )
+    args = p.parse_args(args)
+    errs = []
+    if not args.env:
+        errs.append("env - the KBase environment - is required!")
```

Comment:
It would be helpful to have a list of allowed values here. Is production "prod" or "narrative" or something else.

Actually, after seeing the rest, does this get used? Or is it planned to get used?

Reply:
Good eye!

It is planned but I will delete it for now as it needs better integration into generation of the config file.